org.civicrm.reqangular/src/common/directives/horizontal-scroll-shadows.directive.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-env amd */
+/* globals MutationObserver */
+
+define([
+  'common/angular',
+  'common/modules/directives'
+], function (angular, directives) {
+  'use strict';
+
+  directives.directive('horizontalScrollShadows', [function () {
+    return {
+      restrict: 'A',
+      controller: horizontalScrollShadowsController
+    };
+  }]);
+
+  horizontalScrollShadowsController.$inject = ['$element', '$scope', '$window'];
+
+  function horizontalScrollShadowsController ($element, $scope, $window) {
+    var content, wrapper;
+
+    (function init () {
+      wrapContents();
+      initToggleShadowsHandlers();
+      toggleShadows();
+    }());
+
+    /**
+     * Makes shadow toggle on the wrapper element scrolling,
+     * or DOM changes inside content element, or window resize
+     */
+    function initToggleShadowsHandlers () {
+      wrapper.on('scroll', toggleShadows);
+      angular.element($window).on('resize', toggleShadows);
+      (new MutationObserver(toggleShadows)).observe(content[0], { subtree: true, childList: true });
+      $scope.$watch(function () {
+        return wrapper[0].clientWidth;
+      }, toggleShadows);
+      $scope.$watch(function () {
+        return content[0].clientWidth;
+      }, toggleShadows);
+    }
+
+    /**
+     * Toggles shadows depending on the scroll position
+     */
+    function toggleShadows () {
+      var scroll = wrapper.scrollLeft();
+      var contentWidth = content.width();
+      var wrapperWidth = wrapper.width();
+
+      if (contentWidth <= wrapperWidth) {
+        wrapper.removeClass('insetShadowLeft insetShadowRight');
+
+        return;
+      }
+
+      wrapper[scroll > 0 ? 'addClass' : 'removeClass']('insetShadowLeft');
```

Comment:
what do you like best:

```js
// A:
wrapper[scroll > 0 ? 'addClass' : 'removeClass']('insetShadowLeft');

// B:
(scroll > 0 ? wrapper.addClass : wrapper.removeClass)('insetShadowLeft');

// C:
var classAction = scroll > 0 ? 'addClass' : 'removeClass';

wrapper[classAction]('insetShadowLeft');

// D:
var classAction = scroll > 0 ? wrapper.addClass : wrapper.removeClass;

classAction('insetShadowLeft');
```

Reply:
A.

Reply:
I'm fine with it, tbh but I prefer option B because option A reads like this to me:
`object -> condition -> methodA or methodB -> parameters`

And option B reads:

`condition -> object.methodA or object.methodB -> parameters`

Even if there's repetition the order of object -> method -> parameters is preserved better.

But not a big deal.